analyzer/tests/unit/test_checker_handling.py

```diff
@@ -290,6 +290,24 @@ def setUpClass(cls):
         cls.checks_list = checks.split(',')
         print('Checks list: %s' % cls.checks_list)
 
+    def test_disable_clangsa_checkers(self):
+        """
+        Test that checker config still disables clang-analyzer-*.
+        """
+        analyzer = create_analyzer_tidy()
+        result_handler = create_result_handler(analyzer)
+
+        for arg in analyzer.construct_analyzer_cmd(result_handler):
+            if arg.startswith('-checks='):
+                self.assertIn('-clang-analyzer-*', arg)
+
+        analyzer.config_handler.checker_config = \
+            '{"Checks": "hicpp-use-nullptr"}'
+
+        for arg in analyzer.construct_analyzer_cmd(result_handler):
```

Comment:
Please add one more test case when `take-config-from-directory` is set and `-clang-analyzer-*` can not be found in the argument list. Also add a comment where we check that `take-config-from-directory` is set and we return with a checker list that we do not set `-clang-analyzer-*` consciously.